clusterDeployer.py

```diff
@@ -35,6 +36,76 @@ def match_to_proper_version_format(version_cluster_config: str) -> str:
     return match.group(0)
 
 
+def _update_etc_resolvconf(cluster_name: str, api_vip: str, restore: bool = False) -> None:
+    rc = "/etc/resolv.conf"
+    rc_my = f"/etc/resolv.conf.cda.{cluster_name}"
+    rc_orig = "/etc/resolv.conf.cda-orig"
+    lh = host.LocalHost()
+
+    # The user can disable touching /etc/resolv.conf by
+    #
+    #   $ sudo touch /etc/.resolv.conf.cda-dont-touch
+    #
+    # We will still write our "/etc/resolv.conf.cda*" files, but
+    # it's up to the user to redirect symlinks accordingly.
+    dont_touch = os.path.exists("/etc/.resolv.conf.cda-dont-touch")
+
+    if restore:
+        if dont_touch:
+            return
+        if not os.path.exists(rc_orig):
+            return
+        try:
+            lnk = os.readlink(rc)
+        except Exception:
+            return
+        if not os.path.basename(lnk).startswith("resolv.conf-cda."):
+            # We only restore, if /etc/resolv.conf is a symlink to "resolv.conf-cda.*"
+            # which indicates that the file was created by us.
+            return
+        dst = os.path.basename(rc_orig)
+        if lnk == dst:
+            return
+        logger.info(f"restore /etc/resolv.conf to link to original {rc_orig}")
+        lh.run_or_die(shlex.join(["ln", "-snf", dst, rc]))
+        return
+
+    logger.info(f"updating /etc/resolv.conf to link to {rc_my} (nameserver: {api_vip})")
+    if not os.path.exists(rc_orig):
+        # Backup the original resolv.conf file to "rc_orig"
+        try:
+            with open(rc, "rb") as f:
+                content_rc = f.read()
+        except IOError:
+            pass
+        else:
+            if b"Written by cluster-deployment-automation" in content_rc:
+                pass
+            elif b"Generated by NetworkManager" in content_rc and os.path.exists("/run/NetworkManager/resolv.conf"):
+                # This file seems to be written by NetworkManager. Symlink there.
+                lh.run_or_die(shlex.join(["ln", "-snf", "/run/NetworkManager/resolv.conf", rc_orig]))
+            else:
+                #
+                with common.atomic_write(rc_orig, text=False) as f:
+                    f.write(content_rc)
+    with common.atomic_write(rc_my) as file:
+        file.write(
+            f"""# Written by cluster-deployment-automation for "{cluster_name}"
+# Original file at {rc_orig}
+# Set:     sudo ln -snf {os.path.basename(rc_my)} /etc/resolv.conf
+# Restore: sudo ln -snf {os.path.basename(rc_orig)} /etc/resolv.conf
+nameserver {api_vip}
```

Comment:
Can we also include the nameservers from `/etc/resolv.conf.cda-orig` in `/etc/resolv.conf.cda.<cluster-name>` ?  It's not perfect but at least it kind of keeps connectivity to the outside world working.

Reply:
this is probably not as useful as one might hope. That's because libc will try the nameservers in order, and if the first doesn't work, we hang waiting for timeout. See also `options rotate` in `man resolv.conf`.

This means, DNS is quite broken if the cluster host is unreachable. Which is why a local dnsmasq might be better (but also more complex, so maybe not better after all). Be ready to `sudo ln -snf resolv.conf.cda-orig /etc/resolv.conf` to fix DNS if it's broken.

Anyway. Done.